### DISCUSSION TITLE
Dispersion axis and plane for composite pole figure

### DISCUSSION THREAD

**Discussion by DriptD** [mtex-toolbox/mtex#914] [Ask Anything] [answered]
Hello,

I have a composite pole figure that shows the orientations of <100>,<110>, and <111> - blue, red, and green, respectively,  as shown in the pole figure below - for a set of grains. I am interested in finding the dispersion axis for this particular set of orientations. 

![Comp_PF](https://user-images.githubusercontent.com/77139598/112744959-4e98db80-8fc2-11eb-83df-67ae888ad1d1.jpg)

Here's the code I used to generate the above figure.

```
%% Define Miller Indices
mag100 = Miller(1,0,0,ebsd('Magnetite').CS,'uvw');
mag110 = Miller(1,1,0,ebsd('Magnetite').CS,'uvw');
mag111 = Miller(1,1,1,ebsd('Magnetite').CS,'uvw');

gmag = grains('Magnetite'); % get the grains for magnetite

% Plot the composite pole figure
v100 = ebsd(gmag).orientations*symmetrise(mag100,'unique','antipodal');
plot(v100,'antipodal','Marker','o','MarkerSize',1,'MarkerFaceColor','blue');
hold on
v110 = ebsd(gmag).orientations*symmetrise(mag110,'unique','antipodal');
plot(v110,'antipodal','Marker','o','MarkerSize',1,'MarkerFaceColor','red');
v111 = ebsd(gmag).orientations*symmetrise(mag111,'unique','antipodal');
plot(v111,'antipodal','Marker','o','MarkerSize',1,'MarkerFaceColor','green');
hold off
```

I came across the [fibre.fit](https://mtex-toolbox.github.io/GrainOrientationParameters.html) function and could get the dispersion axis corresponding to the orientations of the individual miller indices but NOT for the composite set (as shown below, Fig. 10 of Till & Moskowitz, 2014). I could not code the same.

![image](https://user-images.githubusercontent.com/77139598/112744954-42ad1980-8fc2-11eb-985b-883228193f0d.png)

1) Is it possible to get the orientation of the 'composite' dispersion axis using [fibre.fit](https://mtex-toolbox.github.io/GrainOrientationParameters.html) ?
2) Is there a way to plot the plane - whose pole the dispersion axis represents - on the pole figure?

I am using MTEX 5.6.1 on Matlab 2020b.

Regards,
Dripta

**Comment by zmichels** [comment, marked as answer]
I wrote some code a while back for finding a best-fit dispersion axis. The scripts are the link below. I can explain later if you want some more info. You can contact me directly at ***@***.***

https://github.com/zmichels/CVA

**Reply by DriptD** [reply]
Thank you very much @zmichels . I got the plot adding the following lines that you suggested.
```
% get the dispersion axis
[ev,mags] = PGA(ebsd(gmag).orientations);
plot(ev(1),'antipodal','Marker','o','MarkerSize',10,'MarkerFaceColor','k')
hold on
plot(ev(1),'plane','linewidth',2)
```

Here it is:
![image](https://user-images.githubusercontent.com/77139598/112757672-6abf6b80-9008-11eb-8f03-9c0a0086197a.png)

Regards,
Dripta

**Comment by kilir** [comment]
Hi,
fibre.fit will also provide you what you show in the figure from Till & Moskowitz. You get the dispersion axis in specimen and crystal coordinates. For relatively "short" fibres it gives the roughly the same as Zachs code.
```
fib = fibre.fit(o)
fib.h % dispersion axis in crystal coordinates
fib.r % dispersion axis in specimen coordinates
```
Cheers,
Rüdiger

**Reply by DriptD** [reply]
Dear Dr. Kilian and Dr. Michels,

Thank you very much.

The orientation of the dispersion axis derived from [fibre.fit](https://mtex-toolbox.github.io/GrainOrientationParameters.html#10) and [PGA()](https://github.com/zmichels/CVA/blob/master/PGA.m) are almost the same. However, for this one dataset, the results are not that close. I used the same code as posted earlier in this thread though.

![image](https://user-images.githubusercontent.com/77139598/112934513-93964c80-913f-11eb-9a82-0fa216b11f56.png)

Any idea what could cause this? Please let me know if you want to take a look at the vector variables corresponding to the three axes. 

Regards,
Dripta

**Comment by zmichels** [comment]
fibre.fit looks like a better fit to me.

**Reply by kilir** [reply]
Looking at your point plot, I'm not sure if one should fit a singe dispersion axis at all. Of course there will be a result, but I wonder what it means if different fibres are mixed.
Cheers,
Rüdiger

**Reply by DriptD** [reply]
Dear Dr. Kilian,

Actually, I'm interested in the slip systems for magnetite grains. I saw Till & Moskowitz (2014) have illustrated a way to do that using a composite plot and thought of trying out the same. The authors, however, have identified the dispersion axes '_visually_' from the stereoplots. They further state, " _the crystal orientations within a large fraction of magnetite grains disperse along small circle girdles around a single dispersion axis when viewed in pole figures_." 

Is there a way to get a 'best-fit' plane upon which the majority of the point clusters (as seen in pole figure) lie? I'm just curious if it's possible.

Regards,
Dripta
